test/test_SW_Flow_Lib.cc

```diff
@@ -308,7 +308,1074 @@ TEST(SWFlowTest, LitterInterceptedWater) {
+
+
+      // Begin TESTs if (totagb < Es_param_limit)
+      totagb = Es_param_limit / 2;
+
+      // Begin TEST if (PET = 0)
+      pot_soil_evap(&bserate, nelyrs, lyrEvapCo, totagb, fbse, petday0, shift,
+        shape, inflec, range, width, swc, Es_param_limit);
+
+      // expect baresoil evaporation rate = 0 if PET = 0
+      EXPECT_DOUBLE_EQ(bserate, 0.) <<
+        "pot_soil_evap != 0 if PET = 0 for " << nelyrs << " soil layers";
+
+
+      // Begin TEST if (potential baresoil rate = 0)
+      pot_soil_evap(&bserate, nelyrs, lyrEvapCo, totagb, fbse0, petday, shift,
+        shape, inflec, range, width, swc, Es_param_limit);
+
+      // expect baresoil evaporation rate = 0 if fbse = 0
+      EXPECT_DOUBLE_EQ(bserate, 0.) <<
+        "pot_soil_evap != 0 if fbse = 0 for " << nelyrs << " soil layers";
+
+
+      // Begin TEST if (swc = 0)
+      pot_soil_evap(&bserate, nelyrs, lyrEvapCo, totagb, fbse, petday, shift,
+        shape, inflec, range, width, swc0, Es_param_limit);
+
+      // expect baresoil evaporation rate = 0 if swc = 0
+      EXPECT_DOUBLE_EQ(bserate, 0.) <<
+        "pot_soil_evap != 0 if swc = 0 for " << nelyrs << " soil layers";
+
+
+      // Begin TEST if (totagb < Es_param_limit)
+      pot_soil_evap(&bserate, nelyrs, lyrEvapCo, totagb, fbse, petday, shift,
+        shape, inflec, range, width, swc, Es_param_limit);
+
+      // expect baresoil evaporation rate > 0
+      // if totagb >= Es_param_limit & swc > 0
+      EXPECT_GT(bserate, 0.) <<
+        "pot_soil_evap !> 0 for " << nelyrs << " soil layers";
+
+      // expect baresoil evaporation rate <= PET
+      EXPECT_LE(bserate, petday) <<
+        "pot_soil_evap !<= PET for " << nelyrs << " soil layers";
+
+      // expect baresoil evaporation rate <= potential water loss fraction
+      EXPECT_LE(bserate, fbse) <<
+        "pot_soil_evap !<= fbse for " << nelyrs << " soil layers";
+
+
+      //Reset to previous global states.
+      Reset_SOILWAT2_after_UnitTest();
+    }
+
+  }
+
+
+  //TEST pot_soil_evap_bs for when nelyrs = 1 and nelyrs = MAX
+  TEST(SWFlowTest, pot_soil_evap_bs){
+    //INPUTS
+    unsigned int nelyrs = 1;
+    double ecoeff[1] = {0.1};
+    double bserate = 0, petday = 0.1, shift = 45, shape = 0.1, inflec = 0.25, range = 0.8;
+    double width[1] = {5};
+    double swc[1] = {1};
+
+    //INPUTS for expected outputs
+    double bserateExpected = 0.06306041;
+
+    //Begin TEST for bserate when nelyrs = 1
+    pot_soil_evap_bs(&bserate, nelyrs, ecoeff, petday, shift, shape, inflec, range, width, swc);
+
+    EXPECT_NEAR(bserate, bserateExpected, 0.0001);  //bserate is expected to be 0.06306041
+
+    //Reset to previous global states.
+    Reset_SOILWAT2_after_UnitTest();
+
+    //Testing when nelyrs = MAX_LAYERS
+    //INPUTS
+    nelyrs = 25;
+    double ecoeff8[25] = {0.1, 0.1, 0.25, 0.5, 0.01, 0.1, 0.25, 0.5, 0.1, 0.1, 0.25, 0.5, 0.01, 0.1, 0.25, 0.5, 0.1, 0.1, 0.25, 0.5, 0.01, 0.1, 0.25, 0.5, 0.4};
```

Comment:
defining `ecoeff8` and `width8` (and to a lesser extent `swc8`) independent from the call to `create_test_soillayers` asks for trouble and unexpected failures if anything changes somewhere else

--> use the approach implemented in `TEST(SWFlowTest, pot_soil_evap)` instead, i.e., extract necessary array values from the soil after the call to `create_test_soillayers`

Reply:
I am unsure how you want me to use the left shift operators and associated comments that you used in the test for pot_soil_evap.  It looks like it should be something like 
`EXPECT_DOUBLE_EQ(bserate,0.) <<`
     `"[function being tested] [conditions to fail the test] for " << [nlyrs] << " soil layers";`
I pushed my updated tests for the pot_evap_bs function and my interpretation of what this part should look like.  If you could elaborate on this bit or tell me not to include it in this set of tests and future tests, I would appreciate it.

Reply:
I was referring to 
```
      ForEachSoilLayer(i) {
        // copy soil layer values into arrays so that they can be passed as
        // arguments to `pot_soil_evap`
        width[i] = s->lyr[i]->width;
        lyrEvapCo[i] = s->lyr[i]->evap_coeff;

        // example: swc as mean of wilting point and field capacity
        swc[i] = (s->lyr[i]->swcBulk_fieldcap + s->lyr[i]->swcBulk_wiltpt) / 2.;
      }
```
as example on how to set up values for `width`, `coeffs`, and `swc`

Reply:
You read up on providing custom text that is printed if an assertion fails here: "To provide a custom failure message, simply stream it into the macro using the << operator, or a sequence of such operators." (https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions)

My comment was not intended this to be the message